File: frontend/src/types/treasury.ts
export enum TransactionType {
  IN = 'IN',
  OUT = 'OUT',
  TRANSFER = 'TRANSFER',
}

export interface BankAccount {
  id: number;
  name: string;
  account_number: string;
  currency: string;
  initial_balance: number;
  current_balance: number;
  is_active: boolean;
}

export interface BankAccountCreate {
  name: string;
  account_number: string;
  currency?: string;
  initial_balance?: number;
}

export interface BankTransactionCreate {
  account_id: number;
  transaction_type: TransactionType;
  amount: number;
  reference?: string;
  description?: string;
  related_entity_type?: string;
  related_entity_id?: number;
}

export interface TransferCreate {
  from_account_id: number;
  to_account_id: number;
  amount: number;
  reference?: string;
  description?: string;
}

export interface WeeklyFixedCostPayload {
  week_reference_date: string;
  admin_payroll: number;
  design_sales_payroll: number;
  production_plant_payroll: number;
  notes?: string | null;
}

export interface WeeklyFixedCostRecord {
  id: number;
  week_reference_date: string;
  admin_payroll: number;
  design_sales_payroll: number;
  production_plant_payroll: number;
  notes: string | null;
  created_by_user_id: number;
  created_at: string;
}

File: backend/app/models/treasury.py
from typing import Optional, List
from sqlmodel import Field, SQLModel, Relationship
from datetime import datetime
import enum

class TransactionType(str, enum.Enum):
    IN = "IN"             # Ingreso
    OUT = "OUT"           # Egreso
    TRANSFER = "TRANSFER" # Transferencia interna

class BankAccount(SQLModel, table=True):
    __tablename__ = "bank_accounts"

    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(index=True)
    account_number: str
    currency: str = Field(default="MXN")
    initial_balance: float = Field(default=0.0)
    current_balance: float = Field(default=0.0)
    is_active: bool = Field(default=True)

    # Relación con transacciones
    transactions: List["BankTransaction"] = Relationship(
        back_populates="account",
        sa_relationship_kwargs={"cascade": "all, delete-orphan"}
    )

class BankTransaction(SQLModel, table=True):
    __tablename__ = "bank_transactions"

    id: Optional[int] = Field(default=None, primary_key=True)
    account_id: int = Field(foreign_key="bank_accounts.id")
    transaction_type: TransactionType
    amount: float
    reference: Optional[str] = None
    description: Optional[str] = None
    transaction_date: datetime = Field(default_factory=datetime.now)
    
    # Rastreabilidad cruzada
    related_entity_type: Optional[str] = None
    related_entity_id: Optional[int] = None

    account: Optional[BankAccount] = Relationship(back_populates="transactions")